design/sketch.webidl

```diff
@@ -401,13 +401,20 @@ dictionary WebGPUShaderModuleDescriptor {
 interface WebGPUShaderModule {
 };
 
-// Description of the framebuffer attachments
-dictionary WebGPUAttachmentsStateDescriptor {
-    sequence<WebGPUTextureFormatEnum> formats;
-    // TODO other stuff like sample count etc.
+// Description of a single attachment
+dictionary WebGPUAttachment {
+    // Attachment data format
+    WebGPUTextureFormatEnum format;
+    // Number of MSAA samples
+    u32 samples;
 };
 
-interface WebGPUAttachmentsState {
+// Description of the framebuffer attachments
+dictionary WebGPUAttachmentsState {
+    // Array of color attachments
+    sequence<WebGPUAttachment> colorAttachments;
+    // Optional depth/stencil attachment
+    WebGPUAttachment? depthStencilAttachment;
```

Comment:
Note important for WebGPU, but this will translate to a nullable pointer to struct in Dawn, which our remoting layer doesn't have support for atm.

Reply:
How else do you express a proper `Option`? :)